core/app/finders/spree/taxons/find.rb

```diff
@@ -0,0 +1,52 @@
+module Spree
+  module Taxons
+    class Find
+      def initialize(taxons, params)
+        @taxons = taxons
+        @params = params
+      end
+
+      def execute
+        taxons = @taxons
+
+        taxons = by_ids(taxons)      if ids?
+        taxons = by_taxonomy(taxons) if taxonomy?
+        taxons = by_roots(taxons)    if roots?
+
+        taxons
+      end
+
+      private
+
+      attr_accessor :params
+
+      def ids?
+        params[:ids].present?
+      end
+
+      def taxonomy?
+        params[:taxonomy_id].present?
+      end
+
+      def roots?
+        params[:roots].present?
```

Comment:
@bbonislawski 
I think we should have convenient methods such as 
```
def roots
  params[:roots]
end
```
or something similar to avoid the redundancy of `params[key]`
We can even define them dynamically if you're okay with it

Reply:
I think that I'd see them as instance variables initialized in constructor like
```
def initialize(taxons, params)
  @taxons = taxons
  @roots = params[:roots]
  @ids = params[:ids].split(',') if params[:ids]
  etc etc
end
```

Reply:
and then maybe private readers for them

Reply:
What do you think?

Reply:
Yes, that's actually the way I'd like finders to be implemented 👍